[jit][await] Apply review comments

IvanKobzarev opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #93284

Differential Revision: [D42849920](https://our.internmc.facebook.com/intern/diff/D42849920)

IvanKobzarev committed:
[jit][await] Apply review comments

[ghstack-poisoned]

IvanKobzarev commented:
@ivankobzarev has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D42849920).

malfet commented on docs/source/index.rst:
```diff
@@ -81,7 +81,6 @@ Features described in this documentation are classified by release status:
    torch.autograd <autograd>
    torch.library <library>
    cuda
-   torch._awaits <_awaits>
```
We can probably leave this one in place as @svekars is working on PR to mark everything that starts with `_` as beta

IvanKobzarev committed:
Update on "[jit][await] Apply review comments"


Differential Revision: [D42849920](https://our.internmc.facebook.com/intern/diff/D42849920)

[ghstack-poisoned]

IvanKobzarev commented:
@ivankobzarev has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D42849920).

IvanKobzarev committed:
Update on "[jit][await] Apply review comments"


Differential Revision: [D42849920](https://our.internmc.facebook.com/intern/diff/D42849920)

[ghstack-poisoned]

IvanKobzarev commented:
@ivankobzarev has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D42849920).

IvanKobzarev committed:
Update on "[jit][await] Apply review comments"


Differential Revision: [D42849920](https://our.internmc.facebook.com/intern/diff/D42849920)

[ghstack-poisoned]

IvanKobzarev commented:
@ivankobzarev has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D42849920).
